.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,58 @@
+# This is a basic workflow to help you get started with Actions
```

Comment:
Thank you for taking your time to look into my work.

I also thought of checking the operativity of the built image and included a simple _Forest_ version check:
[Docker Image Build Check](https://github.com/ChainSafe/forest/runs/5272071920?check_suite_focus=true#step:5:2549)
```
Docker: Service 'forest' Version:
forest v0.2.2/unstable/mainnet/
```

I studied the references in your documentation but I encountered an obstacle for automation which is that the client always requires a password:
```
$ docker exec -it 280c531fdb26 bash
root@280c531fdb26:/# netstat -lpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.1:6116          0.0.0.0:*               LISTEN      1/forest            
tcp        0      0 0.0.0.0:35761           0.0.0.0:*               LISTEN      1/forest            
tcp        0      0 127.0.0.1:1234          0.0.0.0:*               LISTEN      1/forest            
Active UNIX domain sockets (only servers)
Proto RefCnt Flags       Type       State         I-Node   PID/Program name     Path
id3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.DUJiqtAeQJO74xlQ5ySMQj1CqsBgbpqWYrjCojaQpHM:/ip4/127.0.0.1/tcp/1234/http" forest auth api-info -p admin
Enter the keystore passphrase: 

FULLNODE_API_INFO="eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.DUJiqtAeQJO74xlQ5ySMQj1CqsBgbpqWYrjCojaQpHM:/ip4/127.0.0.1/tcp/1234/http"
root@280c531fdb26:/# exit
exit
``` 
according to the _Filecoin_ documentation at
[Create a _FIL_ Wallet](https://docs.filecoin.io/get-started/store-and-retrieve/set-up/#get-a-fil-address)
I created a wallet:
``` 
$ docker exec -e FULLNODE_API_INFO="/ip4/127.0.0.1/tcp/1234/http" -it dda58b640601 forest wallet list
Enter the keystore passphrase: 

$ docker exec -e FULLNODE_API_INFO="/ip4/127.0.0.1/tcp/1234/http" -it dda58b640601 forest wallet new
Enter the keystore passphrase: 

f1fmk5sqz56kqmepjdm35hzw46hcldd5auxh3fudy
$ docker exec -e FULLNODE_API_INFO="/ip4/127.0.0.1/tcp/1234/http" -it dda58b640601 forest wallet list
Enter the keystore passphrase: 

f1fmk5sqz56kqmepjdm35hzw46hcldd5auxh3fudy
```
but I'm always requested for a password.

Also when launching the daemon even when mounting a persistent storage I'm requested for a password to launch the service:
```
$ docker run -v /absolute/path/to/project/.keys/:/root/.forest/:Z -v /absolute/path/to/project/temp/:/var/tmp/:Z -it forest
 2022-02-21T10:59:17.443Z INFO  forest::daemon > Starting Forest daemon, version v0.2.2/unstable/mainnet/
 2022-02-21T10:59:17.443Z INFO  forest_libp2p::service > Recovered libp2p keypair from "/root/.forest/libp2p/keypair"
Enter the keystore passphrase: 
Admin token: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.ERKCoZX4VZXOvbhXVKWqmMGc4BGEHr2FvJNUmcQXExc
```

Is there a way to make this been read from a file?

Reply:
I'm new to the project so maybe there are dedicated flags, but I believe you could just feed forest the contents of your file e.g. `forest < passphrase.txt`? For no phrase at all you could do something along the lines of `echo "" | forest`? Let me know if it works for you.

Reply:
Actually the former daemon state can be recovered using persistent storages at `~/.forest/` and `/var/tmp/` which the **passphrase** introduced from a local file on the docker host:
```
$ docker run -v /absolute/path/to/project/.keys/:/root/.forest/:Z -v /absolute/path/to/project/temp/:/var/tmp/:Z -i forest forest < .keys/passphrase.txt
 2022-02-22T19:31:09.914Z INFO  forest::daemon > Starting Forest daemon, version v0.2.2/unstable/mainnet/
Enter the keystore passphrase:  2022-02-22T19:31:10.007Z INFO  forest_libp2p::service > Recovered libp2p keypair from "/root/.forest/libp2p/keypair"
Admin token: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.ERKCoZX4VZXOvbhXVKWqmMGc4BGEHr2FvJNUmcQXExc
 2022-02-22T19:31:11.447Z INFO  genesis                > Initialized genesis: BlockHeader: Cid("bafy2bzacecnamqgqmifpluoeldx7zzglxcljo6oja4vrmtj7432rphldpdmm2")
 2022-02-22T19:31:11.762Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-0-0-0170db1f394b35d995252228ee359194b13199d259380541dc529fb0099096b0.vk" is ok
 2022-02-22T19:31:11.781Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-0-0-sha256_hasher-ecd683648512ab1765faa2a5f14bab48f676e633467f0aa8aad4b55dcb0652bb.vk" is ok
 2022-02-22T19:31:11.781Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-0-0-sha256_hasher-032d3138d22506ec0082ed72b2dcba18df18477904e35bafee82b3793b06832f.vk" is ok
 2022-02-22T19:31:11.798Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-0-0-7d739b8cf60f1b0709eeebee7730e297683552e4b69cab6984ec0285663c5781.vk" is ok
 2022-02-22T19:31:11.831Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-8-0-559e581f022bb4e4ec6e719e563bf0e026ad6de42e56c18714a2c692b1b88d7e.vk" is ok
 2022-02-22T19:31:11.831Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-0-0-3ea05428c9d11689f23529cde32fd30aabd50f7d2c93657c1d3650bca3e8ea9e.vk" is ok
 2022-02-22T19:31:11.831Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-8-2-b62098629d07946e9028127e70295ed996fe3ed25b0f9f88eb610a0ab4385a3c.vk" is ok
 2022-02-22T19:31:11.831Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-0-0-50c7368dea9593ed0989e70974d28024efa9d156d585b7eea1be22b2e753f331.vk" is ok
 2022-02-22T19:31:11.831Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-0-0-sha256_hasher-6babf46ce344ae495d558e7770a585b2382d54f225af8ed0397b8be7c3fcd472.vk" is ok
 2022-02-22T19:31:11.831Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.vk" is ok
 2022-02-22T19:31:11.840Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-0-0-0cfb4f178bbb71cf2ecfcd42accce558b27199ab4fb59cb78f2483fe21ef36d9.vk" is ok
 2022-02-22T19:31:11.892Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-0-0-5294475db5237a2e83c3e52fd6c2b03859a1831d45ed08c4f35dbf9a803165a9.vk" is ok
 2022-02-22T19:31:11.892Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-2-sha256_hasher-96f1b4a04c5c51e4759bbf224bbc2ef5a42c7100f16ec0637123f16a845ddfb2.vk" is ok
 2022-02-22T19:31:12.270Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-8-2-2627e4006b67f99cef990c0a47d5426cb7ab0a0ad58fc1061547bf2d28b09def.vk" is ok
 2022-02-22T19:31:12.328Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-8-0-0377ded656c6f524f1618760bffe4e0a1c51d5a70c4509eedae8a27555733edc.vk" is ok
 2022-02-22T19:31:15.421Z INFO  paramfetch             > Parameter file "/var/tmp/filecoin-proof-parameters/v28-fil-inner-product-v1.srs" is ok
 2022-02-22T19:31:15.556Z INFO  forest::daemon         > JSON RPC Endpoint at 127.0.0.1:1234
 2022-02-22T19:31:15.578Z INFO  metrics                > Prometheus server started at 127.0.0.1:6116
 2022-02-22T19:31:15.631Z INFO  rpc                    > Ready for RPC connections
```
The previously created _FIL_ wallet is preserved:
```
$ docker exec -e FULLNODE_API_INFO="/ip4/127.0.0.1/tcp/1234/http" -i f43037b47b56 forest wallet list < .keys/passphrase.txt
Enter the keystore passphrase: 
f1fmk5sqz56kqmepjdm35hzw46hcldd5auxh3fudy
```
Important here is that the `passphrase.txt` file must contain the **initial** `passphrase` that was given on the first launch.

Reply:
Seems one can pass `--encrypt-keystore false` to get around this.

Reply:
> Seems one can pass `--encrypt-keystore false` to get around this.

Yes, eventually I had to disable all passphrase testing because the passphrase is obtained through the [`rpassword::read_password`](https://docs.rs/rpassword/latest/rpassword/fn.read_password.html#) function which reads directly from the Command Line and the STDIN is ignored.
[Read Passphrase from Command Line](https://github.com/ChainSafe/forest/blob/cd27985df1ef1469fab70f83d2122c96ac66506a/forest/src/daemon.rs#L73)
```
            let passphrase = read_password().expect("Error reading passphrase");
```

Reply:
I added the container log and the suggested client command which now documents that the container is actually operative.
As documented at:
https://github.com/bodo-hugo-barwich/forest/runs/6146261572?check_suite_focus=true#step:8:60
```
2022-04-24T11:52:23.184Z INFO  rpc                       > Ready for RPC connections
```
but the client report is not that interesting anymore:
https://github.com/bodo-hugo-barwich/forest/runs/6146261572?check_suite_focus=true#step:9:6
```
 2022-04-24T11:52:23.422Z WARN  forest::cli > No configurations found, using defaults.
FULLNODE_API_INFO="***"
```